aten/src/ATen/native/vulkan/VulkanOps.cpp

```diff
@@ -279,6 +274,44 @@ void conv2d_depthwise(
   vkDestroyDescriptorSetLayout(device, descriptorSetLayout, nullptr);
 }
 
+void conv2d_depthwise(
+    VulkanTensor& output,
+    const VulkanTensor& input,
+    const VulkanTensor& weight,
+    const c10::optional<float*> bias,
+    const Conv2DParams params,
+    c10::optional<float> output_min,
+    c10::optional<float> output_max) {
+  conv2d_depthwise(
+      output,
+      input,
+      weight,
+      bufferFromOptionalHostData(bias, conv2d_biasBufferSize(params.OC)),
+      params,
+      output_min,
+      output_max);
+}
+
+void conv2d_depthwise(
+    VulkanTensor& output,
+    const VulkanTensor& input,
+    const float* weight,
+    const c10::optional<float*> bias,
+    const Conv2DParams params,
+    c10::optional<float> output_min,
+    c10::optional<float> output_max) {
+  VulkanTensor weightTensor{{params.OC, params.KH, params.KW}};
+  weightTensor.set_data_from_host(weight);
+  conv2d_depthwise(
+      output,
+      input,
+      weightTensor,
+      bufferFromOptionalHostData(bias, conv2d_biasBufferSize(params.OC)),
+      params,
+      output_min,
+      output_max);
```

Comment:
I recognize this might be difficult to do in eager mode but for performance reasons we want to upload weights and biases to GPU memory ahead of time, and preferably do so with a significant gap, ideally as a one time operation at initialization time.  This usage here is going to act as a sync point otherwise hurting performance.

If weight prepacking is only covering memory reordering and conversion, and weight uploads to GPU memory is still done inline like here, do you envision us having a path to do the aforementioned one-time weight & bias upload?

Reply:
Currently prepacking is folded with https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/prepack_folding.cpp to the start of `forward()` method, (That's my current understanding, I only read it briefly). I think we can play around that folding to move our prepacking to other places of the graph, probably it's loading if it is possible

Reply:
So prepack folding removes all prepacking ops and inserts the outputs of prepacking op as an attribute of the top level module. The attributes are loaded only once so those are done only once.